src/txdb.cpp

```diff
@@ -110,11 +118,19 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
 
     for (CCoinsMap::iterator it = mapCoins.begin(); it != mapCoins.end();) {
         if (it->second.flags & CCoinsCacheEntry::DIRTY) {
-            CoinEntry entry(&it->first);
-            if (it->second.coin.IsSpent())
-                batch.Erase(entry);
-            else
-                batch.Write(entry, it->second.coin);
+            if (it->second.flags & CCoinsCacheEntry::PEGIN) {
+                if (!it->second.peginSpent)
+                    batch.Erase(std::make_pair(DB_PEGIN_FLAG, it->first));
+                else
+                    batch.Write(std::make_pair(DB_PEGIN_FLAG, it->first), '1');
```

Comment:
Note that the value here is a static value, since all we care about is spentness of a peg-in outpoint, and nothing else(no values, scriptpubkeys, etc).